src/mds/topology/topology_item.h

```diff
@@ -490,26 +490,28 @@ class ChunkServer {
           dirty_(false) {}
 
     ChunkServer(ChunkServerIdType id,
```

Comment:
> It is recommended to use the previous method(pass const reference). There also seems to be no performance gain from using the `move constructor`. ago way: `copy constructor` now way: `copy constructor` + `move constructor` And it blurs the promise of const to users.

Even though there's no considerable performance improvement, it can better use move semantics. Because previously when the caller passes rvalues, these values are copied, so that we have to add another overload for rvalues, like,
```cpp
void foo(const std::string& str);
void foo(std::string&& str);
```

But when the number of parameters increases, overload sets become larger, for example, two parameters will introduce four overload functions and 8 for 3 parameters. So far, we haven't considered another scenario - the caller can pass a C-string.

So the recommended way is to take parameters by value and move them into data members, but, still, we should measure the performance, if move operations take a significant amount of time, the overload is a better way.

And, for this case, move operations of std::string shouldn't be a bottleneck.

<hr/>

BTW, @Cyber-SiKu it's better to only apply this method to new parameters.